cyipopt/cython/ipopt_wrapper.pyx

```diff
@@ -654,6 +654,102 @@ cdef class Problem:
 
         return np_x, info
 
+    def get_current_iterate(self, scaled=False):
+        # Allocate arrays to hold the current iterate
+        cdef np.ndarray[DTYPEd_t, ndim=1] np_x
+        cdef np.ndarray[DTYPEd_t, ndim=1] np_mult_x_L
+        cdef np.ndarray[DTYPEd_t, ndim=1] np_mult_x_U
+        cdef np.ndarray[DTYPEd_t, ndim=1] np_g
+        cdef np.ndarray[DTYPEd_t, ndim=1] np_mult_g
+        np_x = np.zeros((self.__n,), dtype=DTYPEd).flatten()
+        np_mult_x_L = np.zeros((self.__n,), dtype=DTYPEd).flatten()
+        np_mult_x_U = np.zeros((self.__n,), dtype=DTYPEd).flatten()
+        np_g = np.zeros((self.__m,), dtype=DTYPEd).flatten()
+        np_mult_g = np.zeros((self.__m,), dtype=DTYPEd).flatten()
+
+        # Cast to C data types
+        x = <Number*>np_x.data
+        mult_x_L = <Number*>np_mult_x_L.data
+        mult_x_U = <Number*>np_mult_x_U.data
+        g = <Number*>np_g.data
+        mult_g = <Number*>np_mult_g.data
+
+        # NOTE: GetIpoptCurrentIterate can *only* be called during an
+        # intermediate callback (otherwise __nlp->tnlp->ip_data_ is NULL)
+        # TODO: Either catch error or avoid calling if we are not in an
+        # intermediate callback
+        ret = GetIpoptCurrentIterate(
+            self.__nlp,
+            scaled,
+            self.__n,
+            x,
+            mult_x_L,
+            mult_x_U,
+            self.__m,
+            g,
+            mult_g,
+        )
+
+        # Return values to user
+        # - Is another data type (e.g. dict, namedtuple) more appropriate than
+        #   simply a tuple?
+        # - Should `ret` be returned here?
```

Comment:
If `ret` is 0, then we should return nan for the values. That is probably easiest to interpret for a Python users.

Reply:
I've raised a RuntimeError in this case. I don't like nan as it's unclear whether it came from Ipopt or us. Return `None`, raise an error, or include `ret` in the returned object are all things I'd be okay with.

Reply:
I was imagining that there could be cases where a problem could complete, but the values weren't returned from these new methods, so you could complete the solve. But maybe that doesn't occur and ret=0 means we should stop the solve.

Reply:
The only case I know of where this returns 0 is if it is called outside of a callback. There are other branches that return false in the Ipopt code, but I don't know how to reach them. I've updated to return `None` in this case so that it's a little easier for a user to handle (i.e. decide whether to complete solve or not).

Reply:
Ok.